DuckDuckGo/AppLifecycle/AppStates/Launched.swift

```diff
@@ -17,12 +17,663 @@
 
+    @UserDefaultsWrapper(key: .didCrashDuringCrashHandlersSetUp, defaultValue: false)
+    private var didCrashDuringCrashHandlersSetUp: Bool
+
+    private let crashCollection = CrashCollection(platform: .iOS)
+    private let bookmarksDatabase = BookmarksDatabase.make()
+    private let marketplaceAdPostbackManager = MarketplaceAdPostbackManager()
+    private let accountManager = AppDependencyProvider.shared.accountManager
+    private let tunnelController = AppDependencyProvider.shared.networkProtectionTunnelController
+    private let vpnFeatureVisibility = AppDependencyProvider.shared.vpnFeatureVisibility
+    private let appSettings = AppDependencyProvider.shared.appSettings
+    private let privacyStore = PrivacyUserDefaults()
+    private let voiceSearchHelper = VoiceSearchHelper()
+    private let autofillLoginSession = AppDependencyProvider.shared.autofillLoginSession
+    private let onboardingPixelReporter = OnboardingPixelReporter()
+    private let widgetRefreshModel = NetworkProtectionWidgetRefreshModel()
+    private let tipKitAppEventsHandler = TipKitAppEventHandler()
+    private let fireproofing = UserDefaultsFireproofing.xshared
+
+    private let vpnWorkaround: VPNRedditSessionWorkaround
+    private let privacyProDataReporter: PrivacyProDataReporting
+    private let isTesting = ProcessInfo().arguments.contains("testing")
+    private let didFinishLaunchingStartTime = CFAbsoluteTimeGetCurrent()
+    private let crashReportUploaderOnboarding: CrashCollectionOnboarding
+
+    // These should ideally be let properties instead of force-unwrapped. However, due to various initialization paths, such as database completion blocks, setting them up in advance is currently not feasible. Refactoring will be done once this code is streamlined.
+    private var uiService: UIService!
+    private var unService: UNService!
+    private var syncDataProviders: SyncDataProviders!
+    private var autoClear: AutoClear!
+    private var syncService: DDGSync!
+    private var isSyncInProgressCancellable: AnyCancellable!
+    private var remoteMessagingClient: RemoteMessagingClient!
+    private var subscriptionCookieManager: SubscriptionCookieManaging!
+    private var autofillPixelReporter: AutofillPixelReporter!
+    private var mainViewController: MainViewController!
+    private var window: UIWindow?
+
+    var urlToOpen: URL?
+    var shortcutItemToHandle: UIApplicationShortcutItem?
+
+    private let application: UIApplication
+
+    // swiftlint:disable:next cyclomatic_complexity
+    init(stateContext: Init.StateContext) {
+
+        application = stateContext.application
+        privacyProDataReporter = PrivacyProDataReporter(fireproofing: fireproofing)
+        vpnWorkaround = VPNRedditSessionWorkaround(accountManager: accountManager, tunnelController: tunnelController)
+        crashReportUploaderOnboarding = CrashCollectionOnboarding(appSettings: AppDependencyProvider.shared.appSettings)
```

Comment:
What is a bit concerning here, is that we change the moment things are instantiated and when data is red from the storage  - and it's potentially earlier than it used to be in the old implementation so things may not be ready yet. E.g. some of the code here for VPN workaround was instantiated lazily in Task {} when did finish launching has already returned to the OS.

Reply:
talked through on 1:1, it shouldn't cause any issues, as there's no work happening in the inits